BEPs/BEP-396.md

```diff
@@ -0,0 +1,182 @@
+	// 1: timely distribution of transaction fees, it will keep partial serial execution when tx cannot delay the distribution
+	Type uint8
+	// Tx Dependency List, the list index is equal to TxIndex
+	TxDeps []TxDep
+	// It indicates the scheduling priority of the transactions
+	SchedulePriority []int
+}
+```
+
+`SchedulePriority` is used to sort transactions by scheduling priority, allowing the scheduler to prioritize the most time-consuming parallel paths, trying to improve the scheduling efficiency under limited CPU resources. This is optional and the default is to sort by TxIndex.
+
+The following shows how to convert the RW read-write relationship graph into a graph structure represented by `TxDAG`.
+
+![](./assets/BEP-396/pic2.png)
+
+### 5.3 TxDAG Validation
+
+Since there is a preset order of transactions in the block, Txi can't depend on Txj (i<j) in DAG, and it is also impossible for circular dependencies to occur. 
+
+The verification rules are as follows:
+
+1. TxIndexes must be smaller than the current TxIndex;
+2. TxIndexes cannot be repeated;
+3. TxIndexes must be sorted in ascending order;
+4. TxIndex in `TxDAG` cannot exceed the maximum TxIndex of the block;
+
+### 5.4 Block Extension
+
+This BEP stores `TxDAG` data in the `BlockBody`, and its Hash in the Extra field of the `Header`. To ensure the correctness of the DAG, if the validator finds an incorrect DAG, it should reject the block.
+
+This is a diagram of the Extra field structure changes in the `Header`:
+
+![](assets/BEP-396/pic3.png)
+
+The specific `BlockBody` structure is defined as follows:
+
+```go
+type Body struct {
+	Transactions []*Transaction
+	Uncles       []*Header
+	Withdrawals  []*Withdrawal `rlp:"optional"`
+	TxDAG        *TxDAG        `rlp:"optional"`
+}
+```
+
+### 5.5 TxDAG Generation
+
+The generation of `TxDAG` requires analysis of the transaction read-write set. You can set a read-write tracker in the EVM's `StateDB`. When reading the state from the DB, it is recorded in the read set. When creating or updating the state, it is recorded in the write set. When reverting the modification, the write set also needs to be reverted.
+
+Finally, the read-write dependency of the transaction is traversed to generate `TxDAG`.
+
+### 5.6 Validating Block
+
+The easiest way to speed up validating blocks through `TxDAG` is to integrate with parallel processors. Usually, parallel processors only optimistically execute transactions in parallel. If `TxDAG` is input, the parallel processor can execute the entire block in the best parallel path. The parallel processor can easily find bad DAG.
+
+At the same time, `TxDAG` can be applied in state prefetcher scenarios, and the state can be loaded in advance through simple parallel prefetchers to improve the cache hit rate of the processor.
+
+![](assets/BEP-396/pic4.png)
+
+If the DAG generated locally by the validator does not match the block, this block should be rejected.
```

Comment:
please add few section about:
1. How to prevent huge and inefficient DAG.
2. How to efficient encode the DAG.

Reply:
updated

Reply:
![image](https://github.com/user-attachments/assets/d19eb5fc-2d06-4ede-b0bb-8f9007209cfc)